### ISSUE TITLE
Lint/RescueWithoutErrorClass triggers for inline rescue

### ISSUE THREAD

**Issue by tomhughes** [bbatsov/rubocop#4833]
## Description

The new `Lint/RescueWithoutErrorClass` cop triggers for inline rescue, but inline rescue doesn't actually allow an error class to be specified. An example report:

```
app/helpers/browse_helper.rb:117:51: W: Avoid rescuing without specifying an error class.
    if page = WIKI_PAGES[locale][type][lookup_us] rescue nil
                                                  ^^^^^^
```

I realise inline rescue probably isn't very good style itself, but it's quite useful here for getting `nil` if any level of the nested hash lookup fails.

## RuboCop version

```
$ rubocop -V
0.50.0 (using Parser 2.4.0.0, running on ruby 2.4.1 x86_64-linux)
```

**Comment by Drenmi** [comment]
Fixed in #4779.

**Comment by Drenmi** [comment]
Also check out [`Hash#dig`](http://ruby-doc.org/core-2.4.2/Hash.html#method-i-dig). 🙂

**Comment by tomhughes** [comment]
Ah cool. I did look this morning to see if there was something like that, but failed to spot it...

**Comment by bbatsov** [comment]
> Also check out Hash#dig.

I see a potential for a new cop. ;-) Unfortunately the potential for false positives I see is also there. :D

**Comment by Drenmi** [comment]
> I see a potential for a new cop. ;-) Unfortunately the potential for false positives I see is also there. :D

I have taken a stab at it before. Can't remember what it was, but something stumped me that time. 🙂
